### PR TITLE
feat: Config auth

### LINKED ISSUE #293: add auth to config command
**Describe the solution you'd like:**
Ability to pass --username and --password to the config command, to authenticate when passing a url

**Why do you want this feature:**
Sometimes a companies config file will be stored in a location that will need authentication to retrieve

### PR DESCRIPTION
**Pull request by seankelly001**
**What this PR does / why we need it**:
Allow users to pass basic auth to config command

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #293

### CODE REVIEW

File: pkg/app/configure.go [hunk 1]
```diff
@@ -110,9 +117,27 @@ func getReader(location string) (io.Reader, error) {
 		if err != nil {
 			return nil, fmt.Errorf("parsing location as URL %s: %w", location, err)
 		}
-		resp, err := http.Get(url.String()) //nolint
+		client := &http.Client{}
+		req, err := http.NewRequestWithContext(ctx, "GET", url.String(), nil)
```
Comment (richardcase) [inline pkg/app/configure.go:121]: ll we create another issue to replace the direct usage of 'net/http' with our mock friendly version?

**Review by richardcase — APPROVED** [review]

File: pkg/app/configure.go (hunk 1 above)
Comment (seankelly001) [reply pkg/app/configure.go:121]: This package here is it? 
https://github.com/fidelity/kconnect/tree/main/pkg/http

Comment (richardcase) [reply pkg/app/configure.go:121]: Thats it. There is already an instance on the App https://github.com/fidelity/kconnect/blob/main/pkg/app/app.go#L49

**Review by richardcase — DISMISSED** [review]
LGTM

**Review by richardcase — APPROVED** [review]

**Pull request closed without merging**